### PR TITLE
Spec improvements

### PR DESCRIPTION
**Pull request by kwxm**
I've 

* Changed the name of the document to plutus-core-specification.

* Added a bib file, plutus-core-specification.bib: if you want to add any new bibliography entries, put them in there.

* Changed from two column text to full width text, in an effort to make it easier to get figures close to relevant text.  At some point I'll go through the document and see if I can adjust the placement of the figures.

* Added a makefile for command-line users: `make` should compile the PDF, making sure that the bibliography and cross-references are correct; `make clean` will get rid of all the intermediate files produced by latex.

* Deleted a lot of commented-out text from an IEEE template that was cluttering up the main file.

### CODE REVIEW

File: plutus-core-spec/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+all: pdf
+
+pdf: ${PDF}
+
+${DOC}.pdf: ${SRC}
+	${LATEX}  ${DOC}
+	${BIBTEX} ${DOC}
+	${LATEX}  ${DOC}   # to make sure the (cross)references are correct
+	${LATEX}  ${DOC}
```
Comment (michaelpj) [inline plutus-core-spec/Makefile]: I *highly* recommend ditching the whole makefile and using `latexmk`, which handles all this stuff for you and is very battle-tested.

**Comment by michaelpj** [comment]
You didn't delete `main.tex`, so it just looks like you've added a new file instead of renaming it.

**Comment by michaelpj** [comment]
(I recommend always giving the diff a once-over before you hit the open PR button, helps a lot to catch things like this!)

File: plutus-core-spec/Makefile (hunk 1 above)
Comment (kwxm) [reply plutus-core-spec/Makefile]: Well yeah, but this is pretty simple.  latexmk takes a lot longer and produces vast amounts of crap.  I'm not going to demand that other people don't use latexmk though.

**Comment by kwxm** [comment]
That's peculiar.  I definitely typed `git mv main.tex plutus-core-spcification.tex`, so I'm not sure how main.tex reappeared.

**Comment by michaelpj** [comment]
I wouldn't normally make a fuss, but getting git to realise it's a rename really helps with tracing the history. Currently it hasn't realised this becasue the introdcution of the new file and the deletion of the old file are in different commits. So please either rebase and squash them together or make sure to squash merge this.

File: plutus-core-spec/Makefile (hunk 1 above)
Comment (michaelpj) [reply plutus-core-spec/Makefile]: If latexmk is doing more work than your makefile then I expect that your makefile is wrong :)

**Comment by michaelpj** [comment]
You need to change the file that `default.nix` builds.

**Comment by kwxm** [comment]
Yes, I just spotted that default.nix needed fixed.

**Comment by michaelpj** [comment]
I did ask if you would squash merge it... never mind.

**Comment by kwxm** [comment]
I'm absolutely certain that I did squash merge it!  When you're merging there's  a big green button that you have to click on, and it definitely said "Squash and Merge" when I closed the PR.  What exactly is the problem?  If I look at the history for the master branch, I only see one commit from this merge.

**Comment by michaelpj** [comment]
So you did, my apologies! I thought it said something different on this page when it's a squash merge but apparently not.

**Comment by kwxm** [comment]
Thank goodness for that!  I thought that whatever went wrong yesterday had come back to haunt me again, but hopefully that's gone for good now.

**Pull request merged**